packages/core-api/src/apis/implementations/auth/ObservableSession.tsx

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { BehaviorSubject } from '../../../lib';
+import { Observable } from '../../../types';
+
+export class ObservableSession<T> {
+  private session: T | undefined;
+  private readonly subject = new BehaviorSubject<T | undefined>(undefined);
+
+  session$(): Observable<T | undefined> {
+    return this.subject;
+  }
+
+  getSession(): T | undefined {
+    return this.session;
```

Comment:
We wanted to expose profile but some providers might not have a profile right? For ex: Github Auth does not implement the profile API. We wanted to observe something that is common to all providers which is `session`. 

Thinking a bit more about it maybe we could only send a state (boolean) if the session is present or not since we don't make use of the session information in the subscribers.